### ISSUE TITLE
Helpers.php ignored if the afterEach contains a call towards Mockery::close() and Patchwork\restoreAll()

### ISSUE THREAD

**Issue by dingo-d** [pestphp/pest#259]
I have an odd issue that I've described in more detail [here](https://github.com/Brain-WP/BrainMonkey/issues/92).

I have several global helpers that I've put in the `Helpers.php` file so that they are always present in all my unit tests (which is what `Helpers.php` should be used if I understood it correctly). They are mostly related to mocking some specific functions for WordPress or doing some global cleanup functionality (deleting a created folder).

I'm using Brain Monkey to help me unit test some WP specific stuff. In order to test WordPress hooks, Brain Monkey has some utilities that need to be added before each test and removed after each test. But it turns out that the `tearDown` method will call `\Mockery::close();` and `\Patchwork\restoreAll();` ([link](https://github.com/Brain-WP/BrainMonkey/blob/9e9baba36763100427e3d828619e535db25569db/inc/api.php#L35-L40)) which seem to kinda remove all the helpers from the `Helpers.php`, and that makes all the tests that depend on them fail.

I'm not sure how or why this happens (not too familiar with how Pest is handling this), but can this be overridden (without actually writing all those patches in every single test)?

I've also noticed that this seems to happen only when the functionality from the Brain Monkey utility is being used. For instance in the test this will pass:

```php
<?php

namespace Tests\Unit\CustomPostType;

use Brain\Monkey;
use EightshiftBoilerplate\CustomPostType\PostTypeExample;

/**
 * Mock before tests.
 */
beforeEach(function () {
	Monkey\setUp();
});

/**
 * Cleanup after tests.
 */
afterEach(function () {
	Monkey\tearDown();
});


test('Test mocks', function() {
	var_dump(get_template_directory());
});

test('Register method will call init hook', function () {
	(new PostTypeExample())->register();

	$this->assertSame(10, has_action('init', 'EightshiftBoilerplate\CustomPostType\PostTypeExample->postTypeRegisterCallback()'));
});
```

And I'll get the correct dump of the `get_template_directory()` that I've mocked in my `Helpers.php` as  

```php
use Brain\Monkey\Functions;

Functions\when('get_template_directory')->justReturn(dirname(__FILE__) . '/data');
```

while if I place the `Test mocks` after the `Register method will call init hook` test, I'll get a failure

```bash
• Tests\CustomPostType\CustomPostTypeExampleTest > Test mocks
   Whoops\Exception\ErrorException

  "get_template_directory" is not defined nor mocked in this test.

  at vendor/brain/monkey/src/Expectation/FunctionStub.php:52
     48▕         );
     49▕      }
     50▕ }
     51▕ PHP;
  ➜  52▕         eval($function);
     53▕     }
     54▕
     55▕     /**
     56▕      * @return string

      +1 vendor frames
  2   tests/CustomPostType/CustomPostTypeExampleTest.php:30
      get_template_directory()
```

I'm unsure what's the root cause of this issue. Brain Monkey and how the patching is set up in that library, or Pest and how `Helpers.php` are handled here 🤷🏼‍♂️

**Comment by dingo-d** [comment]
Just pinging to see if anybody has any clues as to what could cause this behavior?

**Comment by dingo-d** [comment]
I kinda think I understand what is happening. My guess is, since the BrainMonkey `tearDown` method will call

```
        Container::instance()->reset();
        \Mockery::close();
        \Patchwork\restoreAll();
```

as described [here](https://github.com/Brain-WP/BrainMonkey/blob/2f1970536caf46d7ab4482d73399e019d342c5c3/inc/api.php#L35), that will clean up everything for other tests, so nothing I declared in the global `Helpers.php` file will be valid anymore (all the mocks I did using BrainMonkey lib will get cleaned up).

My question is: shouldn't this helper be called on every test? Kinda like a mini global setUp wrapper that runs before the test?

I kinda found a way to circumvent this. I wrapped all my mocks I'm depending on in a function inside the `Helpers.php`, and I'm calling that inside `beforeEach` in all my tests. That seems to be working. Kinda ugly, but does the trick.
